### PR TITLE
Migrate makeAsyncComponent and deferComponentRender to TS

### PR DESCRIPTION
**Pull request by hmhealey**
Something with the newer version of TS I tried using required me to migrate these to TS. The rest of the changes are to satisfy the type checker now that it properly knows the props of deferred and async components.

#### Related Pull Requests
This is part of https://github.com/mattermost/mattermost-webapp/pull/10617

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: components/invitation_modal/index.tsx [hunk 1]
```diff
@@ -93,7 +93,7 @@ export function mapStateToProps(state: GlobalState, props: OwnProps) {
 type Actions = {
     sendGuestsInvites: (teamId: string, channels: Channel[], users: UserProfile[], emails: string[], message: string) => Promise<{data: InviteResults}>;
     sendMembersInvites: (teamId: string, users: UserProfile[], emails: string[]) => Promise<{data: InviteResults}>;
-    sendMembersInvitesToChannels: (channels: Channel[], teamId: string, users: UserProfile[], emails: string[], message: string) => Promise<{data: InviteResults}>;
+    sendMembersInvitesToChannels: (teamId: string, channels: Channel[], users: UserProfile[], emails: string[], message: string) => Promise<{data: InviteResults}>;
```
Comment (M-ZubairAhmed) [inline components/invitation_modal/index.tsx:96]: nice find 🎉

**Comment by M-ZubairAhmed** [comment]
@hmhealey i have QA Deferred, if it still requires QA feel free to change it

**Review by pvev — APPROVED** [review]
LGTM! thanks @hmhealey

**Pull request merged**